simtools/simulator.py

```diff
@@ -0,0 +1,740 @@
+
+        """
+
+        self._simtelSourcePath = Path(cfg.getConfigArg("simtelPath", simtelSourcePath))
+        self._filesLocation = Path(cfg.getConfigArg("outputLocation", filesLocation))
+        try:
+            self._outputDirectory = io.getOutputDirectory(
+                    self._filesLocation, self.label, self.simulator)
+        except FileNotFoundError:
+            self._logger.error("Error creating output directory {}".format(
+                self._outputDirectory))
+            raise
+
+        self._logger.debug(
+            "Output directory {} - creating it, if needed.".format(self._outputDirectory))
+
+    def _loadConfigData(self, configData=None, configFile=None):
+        """
+        Load configuration data
+
+        Parameters
+        ----------
+        configData: dict
+            Dict with simulator configuration data.
+        configFile: str or Path
+            Path to yaml file containing configurable data.
+
+        """
+        configData = gen.collectDataFromYamlOrDict(configFile, configData)
+        if self.simulator == 'simtel':
+            self._loadSimTelConfig(configData)
+        elif self.simulator == 'corsika':
+            self._loadCorsikaConfig(configData)
+
+    def _loadCorsikaConfig(self, configData):
+        """
+        Validate configuration data for CORSIKA shower simulation and
+        remove entries need needed for CorsikaRunner
+
+        Parameters
+        ----------
+        configData: dict
+            Dict with simulator configuration data.
+
+        """
+
+        self._corsikaConfigData = copy(configData)
+
+        try:
+            self.site = names.validateSiteName(self._corsikaConfigData.pop("site"))
+            self.layoutName = names.validateLayoutArrayName(
+                self._corsikaConfigData.pop("layoutName"))
+        except KeyError:
+            self._logger.error("Missing parameter in simulation configuration data")
+            raise
+
+        self.runs = self._validateRunListAndRange(
+            self._corsikaConfigData.pop("runList", None),
+            self._corsikaConfigData.pop("runRange", None)
+        )
+
+        self._corsikaParametersFile = self._corsikaConfigData.pop("corsikaParametersFile", None)
+
+    def _loadSimTelConfig(self, configData):
+        """
+        Load array model and configuration paramters for array simulations
+
+        Parameters
+        ----------
+        configData: dict
+            Dict with simulator configuration data.
+
+        """
+        _arrayModelConfig, _restConfig = self._collectArrayModelParameters(configData)
+
+        _parameterFile = io.getDataFile("parameters", "array-simulator_parameters.yml")
+        _parameters = gen.collectDataFromYamlOrDict(_parameterFile, None)
+        self.config = gen.validateConfigData(_restConfig, _parameters)
+
+        self.arrayModel = ArrayModel(label=self.label, arrayConfigData=_arrayModelConfig)
```

Comment:
I think this should be done in a separate function since this does not load the config, it sets up the model. I know it might require an additional `if` somewhere, but it might be cleaner.

Reply:
Moving it somewhere else would make things much more complicated (as I would have to pass on the arrayModelConfig). To make things clearer, I've renamed the functions to `_load_configuration_and_simulation_model`